text/0000-cfg-path-version.md

```diff
@@ -0,0 +1,484 @@
+
+So far we have only used `path_exists(..)` to refer to paths in the standard 
+library. However, while it will be a less likely use case, you can use the flag
+to test if a path exists in some library in the ecosystem. This can for example
+be useful if you need to support lower minor versions of a library but also
+add support for features in a higher minor version.
+
+## `#[cfg(version("1.27"))]`
+
+Until now, we have only improved our support for library features but never
+any language features. By checking if we are on a certain minimum version of
+Rust or any version above it, we can conditionally support such new features.
+For example:
+
+```rust
+#[cfg_attr(version("1.27"), must_use)]
+fn double(x: i32) -> i32 {
+    2 * x
+}
+
+fn main() {
+    double(4);
+    // warning: unused return value of `double` which must be used
+    // ^--- This warning only happens if we are on Rust >= 1.27.
+}
+```
+
+Another example is opting into the system allocator on Rust 1.28 and beyond:
+
+```rust
+#[cfg(version("1.28"))
+// or: #[cfg(path_exists(::std::alloc::System))]
+use std::alloc::System;
+
+#[cfg_attr(version("1.28"), global_allocator)]
+static GLOBAL: System = System;
+
+fn main() {
+    let mut v = Vec::new();
+    // This will allocate memory using the system allocator.
+    // ^--- But only on Rust 1.28 and beyond!
+    v.push(1);
+}
+```
+
+Note that you won't be able to make use of `#[cfg(version(..))` for these 
+particular features since they were introduced before this RFC's featuers
+get stabilized. However, there will be features in the future to use this
+mechanism on.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `#[cfg(nightly)]`
+
+To the `cfg` attribute , a `nightly` flag is added.
+
+If and only if a Rust compiler considers itself to be on a nightly channel
+it the `nightly` flag be considered active.
+
+## `#[cfg(version("<version_string>"))]`
+
+To the `cfg` attribute, a `version` flag is added.
+This flag requires that a string literal be specified in it inside parenthesis.
+The string literal must have the format:
+
+```
+version_string ::= \d(.\d)?
+```
+
+If and only if a Rust compiler considers itself to have a version which is
+greater or equal to the version in the `version_string` will the
+`#[cfg(version("<version_string>")]` flag be considered active.
+
+## `#[cfg(path_exists($path))]`
+
+To the `cfg` attribute, a `path_exists` flag is added.
+This flag requires that a `path` fragment be specified in it inside parenthesis
+but not inside a string literal. The `$path` must start with leading `::`
+and may not refer to any parts of the own crate (e.g. with `::crate::foo`,
+`::self::foo`, or `::super::foo` if such paths are legal).
+This restriction exists to ensure that the user does not try to
+conditionally compile against parts of their own crate because that crate
+has not been compiled when the `path_exists` flag is checked on an item.
```

Comment:
I was assuming that this would be quite intractable to do due to stage separation.
For dependency crates, we have already compiled them, but the current crate we haven't compiled yet.

I think it would also be pretty footgun-y.. consider:

```rust
#[cfg(path_exists(self::bar)]
fn foo() { .. }

fn bar() { .. }
```

vs.

```rust
fn bar() { .. }

#[cfg(path_exists(self::bar)]
fn foo() { .. }
```